dev/ProgressRing/ProgressRing.xaml

```diff
@@ -15,35 +14,62 @@
         <Setter Property="MinWidth" Value="16" />
         <Setter Property="IsTabStop" Value="False" />
         <Setter Property="Width" Value="32" />
+        <Setter Property="Maximum" Value="100" />
         <Setter Property="Template">
             <Setter.Value>
                 <ControlTemplate TargetType="local:ProgressRing">
                     <Grid x:Name="LayoutRoot" Background="Transparent">
                         <VisualStateManager.VisualStateGroups>
                             <VisualStateGroup x:Name="CommonStates">
-                                <VisualState x:Name="Active">
+                                <VisualStateGroup.Transitions>
+                                    <VisualTransition From="Active" To="DeterminateActive">
+                                        <Storyboard>
+                                            <DoubleAnimation
+                                                Storyboard.TargetName="DeterminateLottiePlayer"
+                                                Storyboard.TargetProperty="Opacity"
+                                                To="1"
+                                                Duration="0:0:0.15" />
+                                            <DoubleAnimation
+                                                Storyboard.TargetName="IndeterminateLottiePlayer"
+                                                Storyboard.TargetProperty="Opacity"
+                                                To="0"
+                                                Duration="0:0:0.15" />
+                                        </Storyboard>
+                                    </VisualTransition>
+                                </VisualStateGroup.Transitions>
+                                <VisualState x:Name="Inactive">
                                     <Storyboard>
                                         <DoubleAnimation
                                             Storyboard.TargetName="LayoutRoot"
                                             Storyboard.TargetProperty="Opacity"
-                                            To="1"
-                                            Duration="0:0:0.15" />
+                                            To="0"
+                                            Duration="0" />
                                     </Storyboard>
                                 </VisualState>
-                                <VisualState x:Name="Inactive">
+                                <VisualState x:Name="DeterminateActive">
                                     <Storyboard>
                                         <DoubleAnimation
```

Comment:
These are 0 duration. Can it be just a setter ?

Reply:
agreed, I think these story boards could all be setters.

---
In reply to: [448373396](https://github.com/microsoft/microsoft-ui-xaml/pull/2584#discussion_r448373396) [](ancestors = 448373396)